pkg/reconciler/v1alpha1/taskrun/taskrun.go

```diff
@@ -379,6 +399,18 @@ func updateStatusFromPod(taskRun *v1alpha1.TaskRun, pod *corev1.Pod) {
 		// update tr completed time
 		taskRun.Status.CompletionTime = &metav1.Time{Time: time.Now()}
 	}
+
+	if taskRunHasOutputImageResource(resourceLister, taskRun) && taskRun.IsSuccessful() {
+		req := kubeclient.CoreV1().Pods(taskRun.Namespace).GetLogs(pod.Name, &corev1.PodLogOptions{Container: imageDigestExporterContainerName})
+		logContent, err := req.Do().Raw()
+		if err != nil {
+			logger.Errorf("Error getting output from image-digest-exporter for %s/%s: %s", taskRun.Name, taskRun.Namespace, err)
+		}
+		err = json.Unmarshal(logContent, &taskRun.Status.ResourcesResult)
+		if err != nil {
+			logger.Errorf("Error getting output from image-digest-exporter for %s/%s: %s", taskRun.Name, taskRun.Namespace, err)
+		}
+	}
```

Comment:
this seems like some logic that could be useful to see broken out into its own function with its own unit tests (and on that note `updateStatusFromPod` could probably be moved into its own package as well, instead of being in this file with the reconciler!)

Reply:
just gonna bump this again - this seems like a pretty substantial chunk of code to put straight into this function 😇

Reply:
thanks! :D